.cirrus.yml

```diff
@@ -157,6 +169,207 @@ task:
     cores_script: src/tools/ci/cores_backtrace.sh freebsd /tmp/cores
 
 
+netbsd_template: &netbsd_template
+  env:
+    CPUS: 2
+    BUILD_JOBS: 3
+    TEST_JOBS: 3
+
+    CIRRUS_WORKING_DIR: /home/postgres/postgres
+    CCACHE_DIR: /tmp/ccache_dir
+
+    PATH: /usr/sbin:$PATH
+    LD_LIBRARY_PATH: $LD_LIBRARY_PATH:/usr/pkg/lib/
+    TCL_CONFIG_PATH: /usr/pkg/lib
+
+    LANG: "C"
+    LC_ALL : "C"
+
+    BSD_CONFIGURE_FEATURES: *BSD_CONFIGURE_FEATURES
+
+  only_if: $CIRRUS_CHANGE_MESSAGE !=~ '.*\nci-os-only:.*' || $CIRRUS_CHANGE_MESSAGE =~ '.*\nci-os-only:[^\n]*netbsd.*'
+
+  compute_engine_instance:
+    image_project: pg-ci-images-dev
+    image: family/pg-ci-netbsd-9-2
+    platform: netbsd
+    cpu: 4
+    memory: 4G
+    disk: 25
+
+  sysinfo_script: |
+    locale
+    id
+    uname -a
+    ulimit -a -H && ulimit -a -S
+    export
+
+  ccache_cache:
+    folder: $CCACHE_DIR
+
+  create_user_script: |
+    useradd postgres
+    chown -R postgres:users /home/postgres
+    mkdir -p ${CCACHE_DIR}
+    chown -R postgres:users ${CCACHE_DIR}
+
+
+task:
+  <<: *netbsd_template
+
+  name: NetBSD - 13 - Autoconf
+  trigger_type: manual
+
+  configure_script: |
+    su postgres <<-EOF
+      ./configure \
+        --enable-cassert --enable-debug --enable-tap-tests \
+        --enable-nls \
+        \
+        ${BSD_CONFIGURE_FEATURES} \
+        \
+        --with-includes=/usr/pkg/include --with-libs=/usr/pgk/lib \
+        CC="ccache cc" \
+        CXX="ccache c++" \
+        CFLAGS="-O0 -ggdb"
+    EOF
+  build_script: su postgres -c "gmake -s -j${BUILD_JOBS} world-bin"
+  upload_caches: ccache
+
+  test_world_script: |
+    su postgres <<-EOF
+      ulimit -c unlimited; \
+      gmake -s ${CHECK} ${CHECKFLAGS} -j${TEST_JOBS}
+    EOF
+
+  on_failure:
+    <<: *on_failure
+
+
+task:
+  <<: *netbsd_template
+
+  name: NetBSD - 13 - Meson
+
+  configure_script:
+    - su postgres -c 'meson setup --buildtype debug -Dcassert=true -Dssl=openssl build'
```

Comment:
Likely would make sense to set a -Duuid=xyz as well (perhaps in other places too)

(Answering individually because github is returning error 500 when starting a review)

Reply:
Uuid tests still fail on openBSD and netBSD, should I set `-Duuid=bsd`?

Reply:
Ah. I forgot about that part. I suggest adding a comment explaining why it's not enabled (both here and for autoconf). And perhaps we can just use -Duuid=e2fs?

Reply:
Done.